task.go

```diff
@@ -0,0 +1,71 @@
+package mesg
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	"github.com/mesg-foundation/core/api/service"
+)
+
+var errOnlyOneOutputKey = errors.New("there should be only one output key in response")
+
+// Task represents a MESG task.
+type Task struct {
+	name    string
+	handler func(*Request) Response
+}
+
+// NewTask creates a task with name, handler executed when a matching task request is received.
+func NewTask(name string, handler func(*Request) Response) Task {
```

Comment:
Is it really useful to have this `NewTask` helper function? The dev could use directly `Task{}` if `name` and `handler` are exported.

Reply:
we have the following now, according to service API definition:
```go
// Task represents a MESG task.
type Task interface {
	Name() Key
	Execute(execution *Execution) (key Key, data Data)
}

func HandleTask(name string, handler func(*Execution) (Key, Data)) Task
```

But, I haven't finalized the names of basic types yet.

Reply:
done